Cilsil/Cil/Parsers/RetParser.cs

```diff
@@ -18,7 +18,7 @@ internal class RetParser : InstructionParser
             switch (instruction.OpCode.Code)
             {
                 case Code.Ret:
-                    var retType = state.Method.ReturnType;
+                    var retType = state.Method.ReturnType.GetElementType();
```

Comment:
This is the whole bug fix. If you don't have GetElementType() the type is "void modreq([System.Runtime]System.Runtime.CompilerServices.IsExternalInit)" rather than just void. The former results in the code not recognizing that this is a void type return method, which results in the crash he reported. 

  .method public hidebysig specialname 
            instance void modreq([System.Runtime]System.Runtime.CompilerServices.IsExternalInit) 
            set_Name(string 'value') cil managed
    {
      .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
      // Code size       8 (0x8)
      .maxstack  8
      IL_0000:  ldarg.0
      IL_0001:  ldarg.1
      IL_0002:  stfld      string ClassLibrary1.TestClass/TestClassRecord::'<Name>k__BackingField'
      IL_0007:  ret
    } // end of method TestClassRecord::set_Name